src/core/server/http/session_storage.ts

```diff
@@ -38,5 +39,5 @@ export interface SessionStorage<T> {
 }
 
 export interface SessionStorageFactory<T> {
-  asScoped: (request: Request) => SessionStorage<T>;
+  asScoped: (request: Readonly<Request> | KibanaRequest) => SessionStorage<T>;
```

Comment:
I see we use `Readonly<Request>` in tests, but would a consumer of the API ever get access to a `Request` to pass it in here?

Reply:
yes, one of the next actions in https://github.com/elastic/kibana/issues/33775 is restrict access to hapi Request object, which is exposed via `registerAuth`  https://github.com/elastic/kibana/pull/34631#discussion_r273135519